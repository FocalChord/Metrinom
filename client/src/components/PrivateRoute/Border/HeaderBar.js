import { AppBar, Button, IconButton, Toolbar } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import React from "react";
import { useHistory, useLocation } from "react-router-dom";
import { CookieManager } from "../../../utils";

const drawerWidth = 260;

const useStyles = makeStyles((theme) => ({
    menuButton: {
        marginRight: theme.spacing(2),
    },
    grow: {
        flexGrow: 1,
    },
    backButton: {
        marginRight: theme.spacing(2),
    },
    forwardButton: {
        marginRight: theme.spacing(2),
    },
    appBar: {
        [theme.breakpoints.up("sm")]: {
            width: `calc(100% - ${drawerWidth}px)`,
            marginLeft: drawerWidth,
        },
    },
}));

const HeaderBar = ({ routes }) => {
    const classes = useStyles();
    const history = useHistory();
    const location = useLocation();
    const isRootRoute = routes.map((i) => i.path).includes(location.pathname);

    return (
        <AppBar className={classes.appBar} position="fixed" color="default">
            <Toolbar>
                <IconButton
                    color="inherit"
                    edge="start"
                    className={classes.backButton}
                    onClick={() => history.goBack()}
                    disabled={isRootRoute}
                >
                    <ArrowBackIosIcon />
                </IconButton>
                <IconButton color="inherit" edge="start" className={classes.forwardButton} onClick={() => history.goForward()}>
                    <ArrowForwardIosIcon />
                </IconButton>
                <div className={classes.grow} />
                <Button
                    color="inherit"
                    onClick={() => {
                        CookieManager.removeUserToken();
                        window.location.assign(window.location);
                    }}
                >
                    Log out
                </Button>
            </Toolbar>
        </AppBar>
    );
};

export default HeaderBar;
