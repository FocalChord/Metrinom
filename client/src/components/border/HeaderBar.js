import { AppBar, Button, IconButton, Toolbar } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import MenuIcon from "@material-ui/icons/Menu";
import React from "react";
import CookieManager from "../../utils/CookieManager";

const drawerWidth = 260;

const useStyles = makeStyles((theme) => ({
    appBar: {
        [theme.breakpoints.up("sm")]: {
            width: `calc(100% - ${drawerWidth}px)`,
            marginLeft: drawerWidth,
        },
    },
    menuButton: {
        marginRight: theme.spacing(2),
        [theme.breakpoints.up("sm")]: {
            display: "none",
        },
    },
    logout: {
        [theme.breakpoints.only("xs")]: {
            display: "none",
        },
    },
    grow: {
        flexGrow: 1,
    },
}));

const HeaderBar = () => {
    const classes = useStyles();

    return (
        <AppBar position="fixed" className={classes.appBar} color="default">
            <Toolbar>
                <IconButton color="inherit" edge="start" className={classes.menuButton}>
                    <MenuIcon />
                </IconButton>
                <div className={classes.grow} />
                <Button
                    color="inherit"
                    className={classes.logout}
                    onClick={() => {
                        CookieManager.removeUserToken();
                        window.location.assign(window.location);
                    }}
                >
                    Log out
                </Button>
            </Toolbar>
        </AppBar>
    );
};

export default HeaderBar;
