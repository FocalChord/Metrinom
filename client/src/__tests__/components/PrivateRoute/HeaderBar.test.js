import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import HeaderBar from "../../../components/PrivateRoute/Border/HeaderBar";

test("<HeaderBar />", () => {
    const tree = renderer.create(<HeaderBar />).toJSON();

    expect(tree).toMatchSnapshot();
});
