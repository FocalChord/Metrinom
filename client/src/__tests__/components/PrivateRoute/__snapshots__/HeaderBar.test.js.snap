// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar /> 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorDefault makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="makeStyles-grow-2"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonLogout-6 MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Log out
      </span>
    </button>
  </div>
</header>
`;
